spot_wrapper/spot_arm.py

```diff
@@ -437,63 +437,73 @@ def gripper_angle_open(self, gripper_ang: float, ensure_power_on_and_stand: bool
 
         return True, "Opened gripper successfully"
 
-    def hand_pose(self, data) -> typing.Tuple[bool, str]:
+    def hand_pose(
+        self,
+        *,
+        x: float,
+        y: float,
+        z: float,
+        qx: float,
+        qy: float,
+        qz: float,
+        qw: float,
+        duration: float = 0.125,
+        ref_frame: str = "body",
+        ensure_power_on_and_stand: bool = True,
+    ) -> typing.Tuple[bool, str]:
         """
         Set the pose of the hand
 
         Args:
-            data:
+            x, y, z: float positions for the pose
+            qx, qy, qz, qw: float quaternion for the pose
+            duration: secodns for the trajectory
+            ref_frame: base frame for the pose. This needs to be something Spot knows about, ie "body" or "arm0.link_sh0"
+            ensure_power_on_and_stand: bool for whether or not to ensure Spot is standing/powered on before executing
 
         Returns:
             Boolean success, string message
         """
         try:
-            success, msg = self.ensure_arm_power_and_stand()
-            if not success:
-                self._logger.info(msg)
-                return False, msg
+            if ensure_power_on_and_stand:
+                success, msg = self.ensure_arm_power_and_stand()
+                if not success:
+                    self._logger.info(msg)
+                    return False, msg
```

Comment:
This is not your doing, but it makes me anxious to see this much code all wrapped up in the same `try` block...

Reply:
I agree, but basically every single function in this module is in a `try... except` to make sure it's not a blocker and always returns I think

Reply:
Agree as well. Especially since its a generic exception that is caught versus a specific one. But since it was already here I think it is ok for now. There is some discuss about some larger changes that will improve much of this.